bazel/emscripten_cache.bzl

```diff
@@ -0,0 +1,103 @@
+BUILD_FILE_CONTENT_TEMPLATE = """
+package(default_visibility = ['//visibility:public'])
+exports_files(['emscripten_config'])
+"""
+
+EMBUILDER_FILE_CONTENT_TEMPLATE = """
+CACHE = '{cache}'
+EMSCRIPTEN_ROOT = '{emscripten_root}'
+BINARYEN_ROOT = '{binaryen_root}'
+LLVM_ROOT = '{llvm_root}'
+
+import platform
+
+system = platform.system()
+machine = "arm64" if platform.machine() in ('arm64', 'aarch64') else "amd64"
+nodejs_binary = "bin/nodejs/node.exe" if(system =="Windows") else "bin/node"
+NODE_JS = '{external_root}/nodejs_{{}}_{{}}/{{}}'.format(system.lower(), machine, nodejs_binary)
+"""
+
+def get_root_and_script_ext(repository_ctx):
+    if repository_ctx.os.name.startswith('linux'):
+        if 'amd64' in repository_ctx.os.arch or 'x86_64' in repository_ctx.os.arch:
+            return (repository_ctx.path(Label("@emscripten_bin_linux//:BUILD.bazel")).dirname, '')
+        elif 'aarch64' in repository_ctx.os.arch:
+            return (repository_ctx.path(Label("@emscripten_bin_linux_arm64//:BUILD.bazel")).dirname, '')
+        else:
+            fail('Unsupported architecture for Linux')
+    elif repository_ctx.os.name.startswith('mac'):
+        if 'amd64' in repository_ctx.os.arch or 'x86_64' in repository_ctx.os.arch:
+            return (repository_ctx.path(Label("@emscripten_bin_mac//:BUILD.bazel")).dirname, '')
+        elif 'aarch64' in repository_ctx.os.arch:
+            return (repository_ctx.path(Label("@emscripten_bin_mac_arm64//:BUILD.bazel")).dirname, '')
+        else:
+            fail('Unsupported architecture for MacOS')
+    elif repository_ctx.os.name.startswith('windows'):
+        fail('Using a secondary cache is not supported on Windows')
+        #return (repository_ctx.path(Label("@emscripten_bin_win//:BUILD.bazel")).dirname, '.bat')
+    else:
+        fail('Unsupported operating system')
+
+def _emscripten_cache_impl(repository_ctx):
+    # Read the default emscripten configuration file
+    default_config = repository_ctx.read(
+        repository_ctx.path(
+            Label("@emsdk//emscripten_toolchain:default_config")
+        )
+    )
```

Comment:
If you can read the default config then the easiest thing to do is just to copy it.. and and a new `CACHE = ` line at the bottom.   Then you should be able to avoid the whole template thing above?

Reply:
That's not really possible. The default configuration relies on the environment variables `EM_BIN_PATH` and `ROOT_DIR` which are only set after the toolchain has been set up